presto-main/src/main/java/com/facebook/presto/server/HttpRequestSessionContext.java

```diff
@@ -107,7 +111,14 @@
     private final String clientInfo;
     private final Map<SqlFunctionId, SqlInvokedFunction> sessionFunctions;
 
+    private final Optional<Tracer> tracer;
+
     public HttpRequestSessionContext(HttpServletRequest servletRequest, SqlParserOptions sqlParserOptions)
+    {
+        this(servletRequest, sqlParserOptions, NoopTracerProvider.NOOP_TRACER);
+    }
+
+    public HttpRequestSessionContext(HttpServletRequest servletRequest, SqlParserOptions sqlParserOptions, Tracer tracer)
```

Comment:
One question: the caller of this method should already know if we need to create an actual tracer or not right? So `isTracingEnabled()` Should be called at the caller level instead of inside the method?

Reply:
Yeah so this is the tricky part. isTracingEnabled() is actually first known when this HttpRequestSessionContext is created, from the session property enable_distributed_tracing. So the caller does not know actually.